module/Finna/src/Finna/AjaxHandler/DynamicList.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Get dynamic list AJAX handler
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2015-2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace Finna\AjaxHandler;
+
+use Vufind\ILS\Connection;
+use VuFind\Record\Loader;
+use Zend\Mvc\Controller\Plugin\Params;
+use Zend\View\Renderer\RendererInterface;
+
+/**
+ * DynamicList AJAX handler
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class DynamicList extends \VuFind\AjaxHandler\AbstractBase
+{
+    /**
+     * Record loader
+     *
+     * @var Loader
+     */
+    protected $recordLoader;
+
+    /**
+     * Connection to ils
+     *
+     * @var Connection
+     */
+    protected $ils;
+
+    /**
+     * View renderer
+     *
+     * @var RendererInterface
+     */
+    protected $renderer;
+
+    /**
+     * Constructor
+     *
+     * @param Loader            $loader   Record loader
+     * @param Connection        $ils      Connection to the ils
+     * @param RendererInterface $renderer View renderer
+     */
+    public function __construct(
+        Loader $loader, Connection $ils, RendererInterface $renderer
+    ) {
+        $this->recordLoader = $loader;
+        $this->renderer = $renderer;
+        $this->ils = $ils;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $query = $params->fromQuery('query', 'mostloaned');
```

Comment:
Tähän tarvitaan lisäksi se, miltä ILS-ajurilta haetaan. Muuten MultiBackend ei pysty sitä määrittämään. Sen pitää olla mukana myös checkFunction-kutsussa (esim. `$result = $this->ils->checkFunction('getDynamicList', [id => "$source.123"]);`, ei tarvitse olla toimiva id) ja tietysti varsinaisessa getDynamicList-kutsussa.

Reply:
Hmm, sourcen otan tuosta $source = $this->getDefaultLoginDriver(); mutta tämä ei taida viitata samaan asiaan?

Reply:
Ei, se on vain jokin mikä nyt sattuu olemaan oletus, eikä välttämättä mikään. Eikä varsinkaan välttämättä Axiell-kirjasto.